doc/forgerock.md

```diff
@@ -0,0 +1,234 @@
+      "e": "AQAB"
+    },
+    {
+      "kty": "EC",
+      "kid": "pZSfpEq8tQPeiIe3fnnaWnnr/Zc=",
+      "use": "sig",
+      "alg": "ES512",
+      "x": "AHdVKbNDHym-MiUh6caaod_ktp8PXN6g1zIKLzlaCSOZP82KKaQsfwltAKnMrw129nVx-2kt8x1J1pp1ADe9HtXt",
+      "y": "AUqhRKcYvA6lElI3UrfqvpuhVsyEFBQ4cM_E9v4WGnRc_priiTVa_UC7YfCtQJT9F8Oc21v_i57Sp3Mq_vw5ueRd",
+      "crv": "P-521"
+    },
+    {
+      "kty": "EC",
+      "kid": "I4x/IijvdDsUZMghwNq2gC/7pYQ=",
+      "use": "sig",
+      "alg": "ES384",
+      "x": "k5wSvW_6JhOuCj-9PdDWdEA4oH90RSmC2GTliiUHAhXj6rmTdE2S-_zGmMFxufuV",
+      "y": "XfbR-tRoVcZMCoUrkKtuZUIyfCgAy8b0FWnPZqevwpdoTzGQBOXSNi6uItN_o4tH",
+      "crv": "P-384"
+    },
+    {
+      "kty": "EC",
+      "kid": "Fol7IpdKeLZmzKtCEgi1LDhSIzM=",
+      "use": "sig",
+      "alg": "ES256",
+      "x": "N7MtObVf92FJTwYvY2ZvTVT3rgZp7a7XDtzT_9Rw7IA",
+      "y": "uxNmyoocPopYh4k1FCc41yuJZVohxlhMo3KTIJVTP3c",
+      "crv": "P-256"
+    }
+  ]
+}
+```
+
+OIDC does not use JWT by default, so it has to be enabled in Services > OAuth2 Provider > Use Stateless Access & Refresh Tokens.
+Then it needs to be configured to use public key cryptography instead of shared secret by selecting any
+RS cypher as "OAuth2 Token Signing Algorithm" in the Advanced tab
+(need to save the previous page before changing the tab).
+
+Create an Application to generate Access Token in Applications > OAuth 2.0 > Add Client.
+Fill all the fields with some bogus values (even the scope and default scope).
+
+Recommended way of getting the Access Token is using Insomnia - cross platform HTTP client.
+Use the "token_endpoint" and "authorization_endpoint" values from the OIDC Discovery endpoint as Authorization and Access Token URL.
+
+In case you get 401 when creating the token terminate Forgerock docker and start again.
+
+After getting JWT Insomnia can make authenticated requests to APIcast. But APIcast needs the Forgerock public key,
+so it can verify JWT signature. The JWK endpoint from OIDC Discovery provides JWK encoded certificates.
+
+Use the one with alg RS256 and convert it to normal RSA certificate:
+Before it can be configured in the policy it needs to be formatted to the one line format like Keycloak uses.
+
+
+APIcast does OIDC Discovery only when downloading configuration from System. Passing configuration file from filesystem won't trigger that process.
+It is possible to mock System API by static files and a simple webserver:
+
+```shell
+gem install adsf
+cd examples/configuration/oidc/
+adsf
+```
+
+And edit `examples/configuration/oidc/admin/api/services/oidc/proxy/configs/production/latest.json` to provide the correct public key.
```

Comment:
I used https://play.golang.org/p/mLpOxS-5Fy to convert the JWK to normal public key.